pkg/storage/ingest/pusher_test.go

```diff
@@ -624,10 +626,129 @@ func TestShardingPusher(t *testing.T) {
 				assert.Equalf(t, tc.expectedErrsCount, receivedErrs, "received %d errors instead of %d: %v", receivedErrs, tc.expectedErrsCount, actualPushErrs)
 			}
 
-			closeErr := shardingP.close()
+			closeErr := shardingP.Stop()
 			assert.ErrorIs(t, closeErr, tc.expectedCloseErr)
 			pusher.AssertNumberOfCalls(t, "PushToStorage", len(tc.expectedUpstreamPushes))
 			pusher.AssertExpectations(t)
 		})
 	}
 }
+
+func TestBatchingQueue(t *testing.T) {
```

Comment:
now that we also added error handling there I think it would make sense to add more tests for that. For example, what happens if the error channel fills up after the queue is already at capacity: any future calls to AddToBatch should return an error until we drain all errors, then AddToBatch should continue working. Also what happens if we push multiple errors after the last AddToBatch: they should all come out when we call Close. What if we push more errors than write requests? 

claude.ai is pretty good at writing these btw 😅